### ISSUE TITLE
Improve handling of Generic Resources

### ISSUE THREAD

**Issue by iocanel** [fabric8io/kubernetes-client#3829]
### Is your enhancement related to a problem? Please describe

The following code:

```
client.load(is).createOrReplace();
```

Can result in:
```
Could not find a registered handler for item: [GenericKubernetesResource ...]
```

However if I construct the ResourceDefinitionContext I can use it like:

```

client.genericKubernetesResources(context).withName(genericResource.getMetadata().getName()).createOrReplace(genericResource);
``` 

### Describe the solution you'd like

Given that I can create the context with a lookup like:

```
    /**
     * Obtain everything the APIResourceList from the server and extract all the info we need in order to know how to create /
     * delete the specified generic resource.
     * 
     * @param client the client instance to use to query the server.
     * @param resource the generic resource.
     * @return an optional {@link ResourceDefinitionContext} with the resource info or empty if resource could not be matched.
     */
    private static Optional<ResourceDefinitionContext> getGenericResourceContext(KubernetesClient client,
            GenericKubernetesResource resource) {
        APIResourceList apiResourceList = client.getApiResources(resource.getApiVersion());
        if (apiResourceList == null || apiResourceList.getResources() == null || apiResourceList.getResources().isEmpty()) {
            return Optional.empty();
        }
        return client.getApiResources(resource.getApiVersion()).getResources().stream()
                .filter(r -> r.getKind().equals(resource.getKind()))
                .map(r -> new ResourceDefinitionContext.Builder()
                        .withGroup(ApiVersionUtil.trimGroup(resource.getApiVersion()))
                        .withVersion(ApiVersionUtil.trimVersion(resource.getApiVersion()))
                        .withKind(r.getKind())
                        .withNamespaced(r.getNamespaced())
                        .withPlural(r.getName())
                        .build())
                .findFirst();
    }
```

I think that it would make sense that the client does this in a transparent manner.

**Comment by iocanel** [comment]
FWIW, the suggested solution is pretty much what `kubectl does`

**Comment by shawkins** [comment]
What version are you using?  This was already implemented in #3404 and #3555

**Comment by manusa** [comment]
https://github.com/quarkusio/quarkus/pull/23031#discussion_r791673977
